cmd/kube-proxy/app/server_others.go

```diff
@@ -175,15 +175,15 @@ func newProxyServer(
 		serviceEventHandler = proxierIPTables
 		endpointsEventHandler = proxierIPTables
 		// No turning back. Remove artifacts that might still exist from the userspace Proxier.
-		klog.V(0).Info("Tearing down inactive rules.")
+		klog.V(0).Info("tearing down inactive rules.")
 		// TODO this has side effects that should only happen when Run() is invoked.
 		userspace.CleanupLeftovers(iptInterface)
-		// IPVS Proxier will generate some iptables rules, need to clean them before switching to other proxy mode.
-		// Besides, ipvs proxier will create some ipvs rules as well.  Because there is no way to tell if a given
-		// ipvs rule is created by IPVS proxier or not.  Users should explicitly specify `--clean-ipvs=true` to flush
-		// all ipvs rules when kube-proxy start up.  Users do this operation should be with caution.
-		if canUseIPVS {
-			ipvs.CleanupLeftovers(ipvsInterface, iptInterface, ipsetInterface, cleanupIPVS)
+		// IPVS proxier will generate some iptable rules, but we should only clean up those rules if cleanupIPVS
+		// is true, otherwise we end up flushing iptable chains that are shared between the IPVS and iptables proxier.
+		// Users should explicitly specify `--clean-ipvs=true` to flush all ipvs rules when kube-proxy starts up.
```

Comment:
also, the command line arg is actually `--cleanup-ipvs`

Reply:
Fixed the command line args. Hard to say if we want to expose this level of detail to users, I personally would not because this is sort of an edge case 🤔 Will defer to your judgement though.